vumi_http_api/vumi_api.py

```diff
@@ -1,23 +1,95 @@
+import base64
+
+from twisted.internet.defer import inlineCallbacks
+from twisted.internet.error import DNSLookupError, ConnectionRefusedError
+from twisted.web.error import SchemeNotSupported
+
 from vumi.application import ApplicationWorker
+from vumi.config import ConfigBool, ConfigInt, ConfigText
+from vumi.utils import http_request_full, HttpTimeoutError
+from vumi import log
+
+from .utils import extract_auth_from_url
 
 
 class VumiApiWorkerConfig(ApplicationWorker.CONFIG_CLASS):
-    pass
+    """ Configuration options for the Vumi API Worker """
+    conversation_key = ConfigText(
+        "Conversation key for the current conversation")
+    push_message_url = ConfigText(
+        "URL for messages to be send to")
+    ignore_messages = ConfigBool(
+        "If True, no messages will be sent to the push_message_url",
+        default=False, static=True)
```

Comment:
We do currently allow `push_message_url` to be `None`? It keeps the old behaviour of a warning level log of `push_message_url not configured for conversation: conversation_id`.